### ISSUE TITLE
Configure loss function in python

### ISSUE THREAD

**Issue by stephen-up** [google/yggdrasil-decision-forests#198]
Hi, I think there is a problem with configuring the loss function in python.

Eg from the docs in the code, it looks like you should be able to configure focal loss like this.
```
model = ydf.GradientBoostedTreesLearner(
    label="label",
    loss="BINARY_FOCAL_LOSS",
).train(df)
```

But no matter what I try, It seems to still be BINOMIAL_LOG_LIKELIHOOD. The eval scores never change and I can see BINOMIAL_LOG_LIKELIHOOD in the model description.

```
Cannot compute model self evaluation:This model does not support evaluation reports.

Loss: BINOMIAL_LOG_LIKELIHOOD
Number of trees per iteration: 1
Node format: BLOB_SEQUENCE
Number of trees: 40
Total number of nodes: 19260
```

I have also tried configuring it through `extra_training_config` but that also doesnt seem to work.

What could be going on. Is there a different way I should be defining it, or is there a bug.

Im using YDF 0.12.0.

**Comment by rstz** [comment]
Hi Stephen,

Thank you for reporting. I cannot reproduce this on Google Colab with YDF 0.12.0
https://gist.github.com/rstz/28467a9f6937f76b77ff893c3c21d5e8
Can you please give some more information or a repro on a public dataset?

**Comment by rstz** [comment]
Ok, so my initial analysis is: This looks like a bug, since YDF is supposed to give the same results for distributed and non-distributed training (modulo small variations in random number generation). This is clearly not the case here.

The model itself reports that it is trained with Binary Focal Loss:
```
# Looking manually at the model header:
from yggdrasil_decision_forests.model.gradient_boosted_trees import gradient_boosted_trees_pb2
dist_focal_model.save("/tmp/dist_focal_model")
gradient_boosted_trees_pb2.Header.FromString(open("/tmp/dist_focal_model/gradient_boosted_trees_header.pb", "rb").read())
```
so the issue seems more low-level than I initially thought. 

I also checked in the debugger that the Binary Focal Loss is indeed called, so I suspect there's some issue with *how* it is called.

@achoum will take a deeper look to find out exactly what's going wrong here.

**Comment by rstz** [comment]
Update on this - I believe the issue is much simpler and it's not a bug 🥳  (though clearly an issue with the documentation).

When using local training, YDF GBTs extract 10% (by default) of the training dataset as validation dataset to use for early stopping and training logs. When using distributed training, no validation dataset is automatically extracted.
The difference you're seeing is therefore a consequence of training on a different dataset. If you set `validation_ratio=0.0` on the local training, the results are very similar (see [gist](https://gist.github.com/rstz/8f628d0c04de3eb625c50a6834808a95)).

I've added a warning about this behavior to the [distributed training tutorial](https://ydf.readthedocs.io/en/latest/tutorial/distributed_training/#limitations-and-tipps).
